cmake/FindSZ.cmake

```diff
@@ -0,0 +1,60 @@
+#------------------------------------------------------------------------------#
+# Distributed under the OSI-approved Apache License, Version 2.0.  See
+# accompanying file Copyright.txt for details.
+#------------------------------------------------------------------------------#
+#
+# FindSZ
+# -----------
+#
+# Try to find the SZ library
+#
+# This module defines the following variables:
+#
+#   SZ_FOUND        - System has SZ
+#   SZ_INCLUDE_DIRS - The SZ include directory
+#   SZ_LIBRARIES    - Link these to use SZ
+#
+# and the following imported targets:
+#   SZ::SZ - The SZ compression library target
+#
+# You can also set the following variable to help guide the search:
+#   SZ_ROOT - The install prefix for SZ containing the
+#              include and lib folders
+#              Note: this can be set as a CMake variable or an
+#                    environment variable.  If specified as a CMake
+#                    variable, it will override any setting specified
+#                    as an environment variable.
+
+if(NOT SZ_FOUND)
+  if((NOT SZ_ROOT) AND (NOT (ENV{SZ_ROOT} STREQUAL "")))
+    set(SZ_ROOT "$ENV{SZ_ROOT}")
+  endif()
+  if(SZ_ROOT)
+    set(SZ_INCLUDE_OPTS HINTS ${SZ_ROOT}/include NO_DEFAULT_PATHS)
+    set(SZ_LIBRARY_OPTS
+      HINTS ${SZ_ROOT}/lib ${SZ_ROOT}/lib64
+      NO_DEFAULT_PATHS
+    )
+  endif()
+
+  find_path(SZ_INCLUDE_DIR sz.h ${SZ_INCLUDE_OPTS})
+  find_library(SZ_LIBRARY NAMES SZ ${SZ_LIBRARY_OPTS})
+  find_library(ZLIB_LIBRARY NAMES zlib ${SZ_LIBRARY_OPTS})
```

Comment:
Is zlib also a dependecy?  If so this should be dropped and instead added via find_package(zlib)

Reply:
After a bit of investigation, it looks like SZ uses their own zlib library and include in their distribution along with SZ. Then, if this is the case, we don't need to use another "find_package(zlib)", right?